sevntu-checks/src/main/java/com/github/sevntu/checkstyle/checks/coding/RedundantReturnCheck.java

```diff
@@ -328,15 +328,25 @@ private static DetailAST findRedundantReturnInCatch(DetailAST lastStatementInCat
         DetailAST currentNodeAst = lastStatementInCatchBlockAst;
         DetailAST toVisitAst = currentNodeAst;
         DetailAST returnAst = null;
-        while (toVisitAst != null) {
+        while (true) {
             toVisitAst = Utils.getNextSubTreeNode(toVisitAst, currentNodeAst);
 
-            if (toVisitAst != null
-                    && (toVisitAst.getParent().getParent().getNextSibling() == null
-                        || toVisitAst.getParent().getParent().getNextSibling().getType()
-                            == TokenTypes.RCURLY)
-                    && toVisitAst.getType() == TokenTypes.LITERAL_RETURN
-                    && toVisitAst.getParent().getNextSibling() == null) {
+            if (toVisitAst == null) {
```

Comment:
why not simply to check children on `return`.
```
$ cat Test.java 
public class Test {
    boolean foo() {
        return true;
    }
}

$ java -jar checkstyle-8.10.1-all.jar -t Test.java  | grep "\[3:" | sed -E "s/ +\| +\|//"
--LITERAL_RETURN -> return [3:8]
   |--EXPR -> EXPR [3:15]
   |   `--LITERAL_TRUE -> true [3:15]
   `--SEMI -> ; [3:19]

```

Reply:
> why not simply to check children on return

While it would give us the same results, to me it seems it would just mask the issue. We should only be scanning the current method we are in. Once we enter an anonymous class, or possible a lambda, we are scanning outside the scope of what we are checking. My changes have the benefit that we will skip those tokens and their children completely, while verify `return` is by itself would still require us to parse the entire anonymous class.

Reply:
Shouldn't this check have lambda as a token?

Reply:
> It force code to have break in code it is the same as multiple exit points from method, or multiple breaks in loops. 

Each loop has 1 break otherwise `SingleBreakOrContinueCheck` would have violated it.

> l dislike and mistrust for while (true)

nested loop `while true` can be changed to:

````
                do {
....
                } while (toVisitAst != null);
````

But we have nothing to make the condition false for code coverage.

1st loop can be changed to:

````
        DetailAST toVisitAst = Utils.getNextSubTreeNode(currentNodeAst, currentNodeAst);
        while (toVisitAst != null) {
....
            toVisitAst = Utils.getNextSubTreeNode(toVisitAst, currentNodeAst);
        }
````

And cover all code coverage.
Do you prefer these more?

Reply:
> Do you prefer these more?

yes, please remove `while (true)`.


>But we have nothing to make the condition false for code coverage

Following code passed UT and coverage on my side:
```
                while (toVisitAst.getNextSibling() == null) {
                    toVisitAst = toVisitAst.getParent();
                }
                toVisitAst = toVisitAst.getNextSibling();
```

Reply:
Done.